### ISSUE TITLE
Add Omni-Bot support for Linux aarch64 (aka RPI/RPIT builds)

### ISSUE THREAD

**Issue by Exagone313** [etlegacy/etlegacy#2539]
Currently, when ET Legacy is installed or built, Omni-Bot binaries only come for other systems/architectures, no binaries for Linux aarch64 are included.

The goal of this issue is to find a way to build Omni-Bot for this architecture and then ship officially built binaries on new installations.

**Comment by Exagone313** [comment]
Relates to:
* https://github.com/jswigart/omni-bot/issues/6
* https://github.com/jswigart/omni-bot/issues/22

**Comment by mittermichal** [comment]
not just RPi, many VPS providers now offer hosting on this architecture

**Comment by rafal1137** [comment]
There is a section about omnibot compilation on aarch64
https://techyian.github.io/2018-10-14-install-enemy-territory-on-raspberry-pi/

**Comment by Exagone313** [comment]
This post misses the Boost installation part, but I got most of the pieces.

My goal is to implement this compilation in Omni-Bot's CI, as official builds come from there.

Since GitHub Actions doesn't have aarch64 runners, I'm trying to cross-compile from x86-64. It's unfortunate that it is required to have libboost built with `-fPIC` to be able to build the omni-bot library, as distributions don't have static libraries compiled with this flag.

I'm first trying a build with the b2 tool from Boost but then I'll modify the CMake files in Omni-Bot repository once it is working.

**Comment by sebdanielsson** [comment]
> This post misses the Boost installation part, but I got most of the pieces.
> 
> My goal is to implement this compilation in Omni-Bot's CI, as official builds come from there.
> 
> Since GitHub Actions doesn't have aarch64 runners, I'm trying to cross-compile from x86-64. It's unfortunate that it is required to have libboost built with `-fPIC` to be able to build the omni-bot library, as distributions don't have static libraries compiled with this flag.
> 
> I'm first trying a build with the b2 tool from Boost but then I'll modify the CMake files in Omni-Bot repository once it is working.

If it's possible to build under macOS, GitHub offers arm64 runners since January.
https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/

**Comment by Exagone313** [comment]
I successfully cross-built omni-bot for aarch64 (using an x86-64 host), see the steps I used on https://github.com/jswigart/omni-bot/issues/22#issuecomment-2068260272

I had some issues so editing CMake and GitHub actions on omni-bot repository won't be as easy as I wanted.

**Comment by rafal1137** [comment]
@Exagone313 Looks like its done/fixed. Do you want to close this issue by yourself ?

**Comment by Exagone313** [comment]
@rafal1137 Only a part is done, this issue will be completed once ET legacy can ship omni-bot for Linux aarch64.

**Comment by rafal1137** [comment]
aarch omnibot file should be included in atleast latest snapshot build
This issue can be closed

**Comment by Exagone313** [comment]
@rafal1137 I don't think it does:
```
$ curl -LO https://mirror.etlegacy.com/omnibot/omnibot-linux-latest.tar.gz
$ tar tf omnibot-linux-latest.tar.gz | grep 'so$'
omni-bot/omnibot_et_mac.so
omni-bot/omnibot_et.so
omni-bot/omnibot_et.x86_64.so
```

The binaries included in the archive are from [omni-bot latest release ET archive](https://github.com/jswigart/omni-bot/releases/download/0.9/omni-bot_0_90_ET.zip) which doesn't contain Linux aarch64 binaries since it depends on builds made from the CI of omni-bot repository.

It'll be fixed when I (or someone else) will have modified omni-bot CI and then convinced the omni-bot maintainer to make a new release with this CI :smile: 

Then the new official binary could be included in the ET Legacy omni-bot mirror archives and next ET Legacy releases will be able to ship with Linux aarch64 omni-bot.

**Comment by plastovicka** [comment]
I've already added file **omnibot_et.aarch64.so** to the [Omni-bot SVN repository](http://svn.assembla.com/svn/omnibot/Enemy-Territory/0.8/). Can you add it to etlegacy installer ? Then you can also delete the message "No Omni-Bot server support!" from the [download](https://www.etlegacy.com/download) page.

**Comment by Exagone313** [comment]
I'm still a bit unsure if it was so wise to add my build in those archives. If someone has more knowledge in the build issues I spoke of in [my comment](https://github.com/jswigart/omni-bot/issues/22#issuecomment-2068260272), it would be nice to update omni-bot's own CI and make new builds with GitHub Actions where build hashes could be more transparent.

**Comment by rafal1137** [comment]
> I've already added file **omnibot_et.aarch64.so** to the [Omni-bot SVN repository](http://svn.assembla.com/svn/omnibot/Enemy-Territory/0.8/). Can you add it to etlegacy installer ? Then you can also delete the message "No Omni-Bot server support!" from the [download](https://www.etlegacy.com/download) page.

There is `ETLInstallOmniBot.cmake` that downloads zipped/packed omnibots from 
```
message(STATUS "Installing Omni-Bot")
# Note: used archive (20MB) doesn't contain incomplete nav- and other unwanted files
if(UNIX AND NOT APPLE)
	set(ETLEGACY_OMNIBOT_ARCHIVE "omnibot-linux-latest.tar.gz")
	set(ETLEGACY_OMNIBOT_ARCHIVE_URL "https://mirror.etlegacy.com/omnibot/omnibot-linux-latest.tar.gz")
elseif(APPLE)
	set(ETLEGACY_OMNIBOT_ARCHIVE "omnibot-macos-latest.tar.gz")
	set(ETLEGACY_OMNIBOT_ARCHIVE_URL "https://mirror.etlegacy.com/omnibot/omnibot-macos-latest.tar.gz")
elseif(WIN32)
	set(ETLEGACY_OMNIBOT_ARCHIVE "omnibot-windows-latest.zip")
	set(ETLEGACY_OMNIBOT_ARCHIVE_URL "https://mirror.etlegacy.com/omnibot/omnibot-windows-latest.zip")
else()
	message(WARNING "Platform not supported for Omni-Bot")
	return()
endif()
```
Those latest downloads are handled by our backed/server I need to contact @morsik or @jackeri

**Comment by plastovicka** [comment]
Can you also fix a bug in [g_etbot_interface.cpp](https://github.com/jswigart/omni-bot/commit/e483b095666c20fecf88ce18cd6d987830e3796e#diff-350e3d3ef74db4f49e8e7da6033931541474fe2af2a3d20b242966633c83fd31) ? It's necessary to cast to (signed char), because (char) is unsigned on ARM. Bots can't move left and are often killing themselves.

**Comment by plastovicka** [comment]
File **omnibot_et.aarch64.so** is missing in the installer of ET:Legacy 2.84.0.

**Comment by Vorschreibung** [comment]
Did this ever exist? I might remember wrongly but AFAIK Omnibot can't be compiled to aarch64; please correct me if I'm wrong

**Comment by Exagone313** [comment]
@Vorschreibung there is a build in the latest release of https://github.com/jswigart/omni-bot/releases (`omni-bot_0_93_ET.zip`)

```bash
$ unzip -l omni-bot_0_93_ET.zip | grep 'so$'
  3382136  2026-04-14 18:30   omni-bot/omnibot_et.aarch64.so
  3468300  2026-04-14 15:14   omni-bot/omnibot_et.so
  3317136  2026-04-14 15:16   omni-bot/omnibot_et.x86_64.so
  5247584  2026-04-14 16:58   omni-bot/omnibot_et_mac.so
```

edit to plastovicka: oh you are the maintainer, sorry

**Comment by Vorschreibung** [comment]
Gotcha, we are fetching from a mirror of ours; AFAIS this needs to be updated/synced on our side.

**Comment by morsik** [comment]
Yeah, seems like we're still using old assembla SVN and 0.8 version in updater script...

**Comment by morsik** [comment]
I have updated script on https://mirror.etlegacy.com/omnibot/ to point onto new GitHub omnibot repo.

Now it downloads latest release zip file (original GitHub release), repacks onto new files that contains only what installed needs, and creates `-latest` symlink to point into latest repacked files.

It turned out, we were having old omnibot 0.8 on that mirror. Now it's really latest 0.93.

@plastovicka I hope you don't mind doing full mirror like that ;)
Please verify installation again for the ARM.

I'll close this issue once it's confirmed it works.

**Comment by plastovicka** [comment]
You should also update "AArch64 installer" and "AArch64 archive" on page [https://www.etlegacy.com/download](https://www.etlegacy.com/download). And then delete message "No Omni-Bot server support!".
